internal/testrunner/runners/pipeline/ingest_pipeline.go

```diff
@@ -0,0 +1,219 @@
+package pipeline
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/elastic/elastic-package/internal/packages"
+)
+
+const defaultPipelineName = "default"
+
+var ingestPipelineTag = regexp.MustCompile("{{\\s*IngestPipeline.+}}")
+
+type pipelineResource struct {
+	name    string
+	format  string
+	content []byte
+}
+
+type simulatePipelineRequest struct {
+	Docs []pipelineDocument `json:"docs"`
+}
+
+type pipelineDocument struct {
+	Source json.RawMessage `json:"_source"`
+}
+
+type simulatePipelineResponse struct {
+	Docs []pipelineIngestedDocument `json:"docs"`
+}
+
+type pipelineIngestedDocument struct {
+	Doc pipelineDocument `json:"doc"`
+}
+
+func installIngestPipelines(esClient *elasticsearch.Client, datasetPath string) (string, []pipelineResource, error) {
+	datasetManifest, err := packages.ReadDatasetManifest(filepath.Join(datasetPath, packages.DatasetManifestFile))
+	if err != nil {
+		return "", nil, errors.Wrap(err, "reading dataset manifest failed")
+	}
+
+	nonce := time.Now().UnixNano()
+
+	mainPipeline := getWithPipelineNameWithNonce(getPipelineNameOrDefault(datasetManifest), nonce)
+	pipelines, err := loadIngestPipelineFiles(datasetPath, nonce)
+	if err != nil {
+		return "", nil, errors.Wrap(err, "loading ingest pipeline files failed")
+	}
+
+	jsonPipelines, err := convertPipelineToJSON(pipelines)
```

Comment:
I suspect that the format conversion may introduce some difficulties, but this is the format supported by the library. I'm not sure if I can switch to JSON (if library internals allows for this).

Reply:
Probably also something that can be done by the system tests to load it as yaml to keep this part simple.

Reply:
System tests will not be directly loading any ingest pipelines. The will be relying on Kibana (Ingest Manager) to do this as it would in actual usage.